File: Manuel/2025-07-31/나머지/remainder.py
import sys

def solve():
    # set은 중복되는 값을 자동으로 제거함
    remainders = set()

    for _ in range(10): # 10번 동안
        num = int(sys.stdin.readline()) # 정수 입력받기
        remainder = num % 42 # 입력받은 정수를 42로 나눈 나머지 구하기
        remainders.add(remainder) # set에 값 추가하기
    
    print(len(remainders)) # 중복 제거된 set의 값 개수 출력

solve()

File: Manuel/2025-07-31/나머지/remainder.cpp
#include <iostream>
#include <unordered_set> // 파이썬의 set과 유사한 자료형을 쓰기 위함
#define endl "\n"

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    unordered_set<int> remainders; // 중복되는 수를 자동 제거하는 자료형

    for(int i = 0; i < 10; i++)
    {
        int num;
        cin >> num;
        int remainder = num % 42;
        remainders.insert(remainder);
    }

    cout << remainders.size() << endl;

    return 0;
}